rfcs/0018-glides-trainsheet-events.md

```diff
@@ -141,15 +106,37 @@ In order to reduce event duplication, a Trip Key is used to identify both added
 
 ## Events
 
+### Editors Changed
+Operator sign in and out of editing specific trainsheets. This event publishes changes to who is editing. Since one action by an inspector might result in multiple changes, for example if they sign out of one location and in to another, the changes are published in a list.
+
+In practice, we limit each location to have only one editor, and limit each editor to only one location, but that rule may or may not change and this data in this event doesn't guarantee that.
+
+Note that the author may or may not match the editors in `changes`, since inspectors can change whether other people are editing. For example, when an inspector takes over editing at a location, they may sign out the previous editor. Then they would issue a `stop` change for the previous editor and a `start` change for themself.
+
+Event type: `com.mbta.ctd.glides.editors-changed.v1`
+Fields in the event:
+- `author` (Author): the inspector who made the changes
+- `changes` (array of EditorChange): a list of start and stop editing events
+
+#### EditorChange
+Fields in the object:
+- `type` (string `"start"`|`"stop"`): Whether the editor started or stopped editing.
+- `location` (Location): the location the editor started or stopped managing.
+- `emailAddress` (string): the e-mail of the editor.
+- `badgeNumber` (string, optional): the badge number of the editor.
+
+
 ### Operator signed in
 At the start of their shift, operators need to confirm that they are fit-for-duty and do not have any electronic devices. They currently do this by physically signing a paper trainsheet: in the future, they will do this digitally.
 
 Event type: `com.mbta.ctd.glides.operator_signed_in.v1`
 Fields in the event:
 - `author` (Author): the inspector who signed in the operator
-- `operator` (Operator): the operator who signed in
+- `operator` (string): the badge number of the operator who signed in
 - `signedInAt` (RFC3999 timestamp): the time at which they signed in (separate from the `time` of the event)
-- `confirmation` (object): how the operator confirmed that they signed in. Some possibilities are their badge number (`{"type": "<badge number>"`}) or an RFID tag number (`{"rfid": "<RFID tag">}`). The specific formats may change: consumers SHOULD NOT depend on any specific format.
+- `confirmation` (string): how the operator confirmed that they signed in. Current possibilities are `"type:<badge number>"` if they signed by typing in their badge number, or `"tap"` if they signed in by tapping their badge on an RFID reader, but future string formats may be added without warning, and consumers SHOULD NOT depend on any specific format.
```

Comment:
Oh, yeah. Melody and I talked about it, and realized that we didn't actually want to send the RFID serial, because that alone is enough to impersonate someone's physical badge. (We will still start storing the tapped serial number in our own DB.)

And unlike `droppedReason`, there's no human-entered text here, so it'll be easy to add any future information as part of the string, and still have it be parseable. This is even how we store it in our own DB.

Reply:
Changed back to an object, with a slightly different schema. (Just a `type` field, removing the typed data, since it didn't carry any new information anyway.)